### ISSUE TITLE
$PYTHONPATH in notebook is different than in terminal

### ISSUE THREAD

**Issue by blaylockbk** [microsoft/vscode-jupyter#9436]
## Environment data

-   VS Code version: 1.65.2
-   Jupyter Extension version (available under the Extensions sidebar): v2022.2.1030672458
-   Python Extension version (available under the Extensions sidebar): v2022.2.1924087327
-   OS (Windows | Mac | Linux distro) and version: Linux
-   Python and/or Anaconda version: Python 3.9.10
-   Type of virtual environment used: conda
-   Jupyter server running: Local

## Expected behaviour

$PYTHONPATH should be the same between a notebook and the integrated terminal.

## Actual behaviour

After restarting vscode and opening a fresh jupyter notebook, when I run 
```
%%bash
echo $PYTHONPATH
```
in a notebook cell, the variable value is _different_ than when I run the same command in the integrated terminal
```
echo $PYTHONPATH
```

Furthermore, I opened the notebook in JupyterLab and ran the cell and the $PYTHONPATH variable matches the value of the integrated terminal.

Summary: $PYTHONPATH variable matches between terminal and JupyterLab, but not in the vscode notebook.

## Steps to reproduce:

Sorry, I'm not sure how to reproduce this.

I did change the value of the PYTHONPATH variable a few days ago. Is it possible that vscode notebook is using a cached value of PYTHONPATH that needs to be reset?

Thanks for any help you can give.

**Comment by DonJayamanne** [comment]
Thanks for filing this issue, please could you provide the following information:
* Value of PYTHONPATH in your integrated terminal
* Value of PYTHONPATH in VS Code

Also, please provide all of the output from the `Jupyter` output panel so we can see what's going on (we can see more information on the kernel started, paths altered, and the like)

**Comment by DonJayamanne** [comment]
THanks for the information, do you have any values defined in an environment variable or .env file for `PYTHONPATH`?

**Comment by blaylockbk** [comment]
yes, the PYTHONPATH is set in my .bashrc file

**Comment by DonJayamanne** [comment]
That explains it, when we launch the kernels, the extension does not read bashrc files. In fact we don't read any shell scripts the user may have configured. To load variables from a shell script we'd need to initialize a shell, and this has a number of problems associated with that (it can be very slow, sometimes even slow down VS Code completely and cause perf issues).

I'm sorry, however at this stage I don't see a way around this, except for adding the PYTHONPATH path value into your settings.json as follows:
* Create a .env file in a workspace folder that you open  (add the value for PYTHONPATH into that file `PYTHONPATH=xyz`)
* The extension will automatically load env variables from this .env file

Optionally, if you would like this to apply to all workspace folders, then please do file an issue in this repo and we'll look into that.
Basically today you need a .env file per workspace, if you'd like to share that across all workspaces we don't support that, but should be easy to do (as the Python extension supports that).


Closing this issue, as we're not going to support reading the shell scripts and the work around is to  use a .env file.

**Comment by blaylockbk** [comment]
Thanks for your help. I'll make the necessary changes.
